### ISSUE TITLE
Auto Analysis feature is not working

### ISSUE THREAD

**Issue by roopalN** [reportportal/reportportal#1567]
Background : Both service analyzer and elastic service are running. I also started the auto analysis in reportportal user interface.

However, Generate and remove index buttons/features are disabled.

![image](https://user-images.githubusercontent.com/25950710/139455787-8b5dc2b8-c5f5-4755-8522-dcdfd0299ccd.png)

After generating some constant defects ( having similar failure cause and decisions), auto-analysis is not resulting the expected behavior. Rather on refresh lable (auto-analysis  ) is disappearing.

![image](https://user-images.githubusercontent.com/25950710/139455574-bbbcd3d6-2238-41e2-84ac-338c9c97d424.png)

**Comment by satishkumar9549** [comment]
We are also facing the same issue, Can some one help me with this?

**Comment by evjlobanova** [comment]
Hello @satishkumar9549 @roopalN 
Please check our documentation https://reportportal.io/docs/Resolve-Analyzer-Known 
It should help. If you still meet the issue, please let me know
cc @Cyberglamdring

**Comment by roopalN** [comment]
> Hello @satishkumar9549 @roopalN Please check our documentation https://reportportal.io/docs/Resolve-Analyzer-Known It should help. If you still meet the issue, please let me know cc @Cyberglamdring

@evjlobanova we have verified the issues mentioned under known issues already and setup is fine, services are running. Yet , generate index buttons are disabled and auto-analysis is idle.

**Comment by evjlobanova** [comment]
@roopalN could you please send us  logs from the service analyzer 
@Cyberglamdring Hleb, which information do we also need?

**Comment by Cyberglamdring** [comment]
@roopalN, please provide analyzer logs. We need to make sure Analyzer is connected to rabbitmq, elasticsearch and minio

**Comment by roopalN** [comment]
@Cyberglamdring here is the snapshot post trying to auto analyze 1 launch : 
![image](https://user-images.githubusercontent.com/25950710/141287066-b1c2ad6f-97c6-4b4d-85e9-a4fb26c6b171.png)
 let me know, if you need it in different format.

**Comment by roopalN** [comment]
> @roopalN, please provide analyzer logs. We need to make sure Analyzer is connected to rabbitmq, elasticsearch and minio

here are some errors noticed in logs: 
![image](https://user-images.githubusercontent.com/25950710/141287998-81d7c1c0-151d-43f3-acbb-7f53299fbb28.png)
 Can you give pointers for debugging?

**Comment by Cyberglamdring** [comment]
@roopalN, Analyzer doesn't have permissions to ElasticSearch:
![image](https://user-images.githubusercontent.com/50049166/141294449-04ba8980-7fe5-4376-80ea-1d13681c70c4.png)

Check:
1. Availability of ElasticSearch, for example 'curl http//elastic:password@ESHOST:9200' from RP VM
2. Credentials.

**Comment by roopalN** [comment]
@Cyberglamdring Thanks for the direction, we double checked and found that mentioned configs/creds are intact. However, index 34 related error is also shown up, what can it be? Can it be the reason for analysis failure. Any relevant table we should be looking into DB?
![image](https://user-images.githubusercontent.com/25950710/141463393-69c81517-98f9-45e6-b413-0561db1ad252.png)

P.S I removed attached logs as it contained confidential info. If possible, can you also remove the quoted one?

**Comment by Cyberglamdring** [comment]
The index is stored in ElasticSearch. Are the analyzer buttons active? Try to regenerate Index

**Comment by roopalN** [comment]
> The index is stored in ElasticSearch. Are the analyzer buttons active? Try to regenerate Index

We are able to enable Analyzer but index buttons are disabled. Refer the very first screen-shot in the thread. We are suspecting, this might be causing non-functioning of this feature? How to enable these buttons or if you can share troubleshoot points ?

**Comment by evjlobanova** [comment]
The analyzer is working successfully.
The root cause can be in the analyzer deployment

**Comment by cailyoung** [comment]
We are also seeing this happen on a Kubernetes deployment of v5.7.3

I have already checked logs in the `api` and `analyzer` pods and they both seem to be quite happily talking to RabbitMQ and Elasticsearch, just the UI is convinced that the analyzer is unavailable.

Can you help me understand _how_ it is checking that the analyzer is working? I had a look in the code and it seems like it's checking for specific exchange(s) in the rabbitMQ vhost, but beyond that I'm quite confused.

**Comment by cailyoung** [comment]
Update: We identified the problem; we are not using nginx as ingress, rather our own layer 7 app gateway - and we had applied a path rewrite to the entire Ingress. There are API calls to `/composite/info` that are meant to hit the `index` service but due to our rewrite were being passed in as requests to `/info` on the index service, which succeeds, but returns a payload that other parts of the application are not expecting, so they think that the service is down.

We resolved this by splitting the Ingress into one for the `/` path prefix, and another for the other path prefixes.
